src/app/Http/Middleware/PollPermsViewer.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Closure;
+use App\Presentation;
+use App\Poll\State;
+use Illuminate\Http\Response;
+
+class PollPermsViewer
+{
+    /**
+     * Handle an incoming request.
+     *
+     * Only allow admins or viewers when poll is not a draft
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        // determine presentation base on request content
+        $is_poll = $request->poll != null;
+        $presentation = $request->presentation;
+        if ($presentation == null) {
+            $presentation = $request->poll->presentation;
+        }
```

Comment:
why not do a `$presentation = $request->poll->presentation` directly instead of testing if it exists as $request->presentation ?

Reply:
This is because in `api.php` this middleware handles routes which can have either a poll or a presentation.

Reply:
why you handle presentations on a middleware called PollPermsViewer ?
each middleware should do one specific thing

Reply:
Fixed as per 734eedb